### DISCUSSION TITLE
dimension assigned to 'crs' variable in single vs. multiple open

### DISCUSSION THREAD

**Discussion by mdsumner** [pydata/xarray#7035] [General] [answered]
With these NetCDF files there is a dummy ‘crs’ variable that exists to store projection info for the data vars, as per CF conventions for a `grid_mapping` attribute to nominate a  crs.

When xarray reads one dataset with `open_dataset()` it correctly sits on its own unused scalar dimension, but with `open_mfdataset()` crs is on the ‘time’ dimension. This causes problems with append, I'm new to this but I don't expect that crs var to live on the time dimension, I think it's a mistake. 

I've created a notebook, with wget calls to download example files for a reprex.  The notebook uses xarray 2022.3.0, but I tested as well with latest (xarray: 2022.6.1.dev60+g63ba862d). 

https://github.com/mdsumner/xarray_vrt_expt/blob/master/issues/crs_time.ipynb

The files are [NSIDC 25km south polar sea ice concentrations](https://n5eil01u.ecs.nsidc.org/PM/NSIDC-0051.002/), newly released to replace the old binary files.

**Comment by keewis** [comment, marked as answer]
By default, `xarray` will concatenate all variables regardless of whether they have the concat dimension, but that's customizable using the `data_vars` and `coords` parameters.

Does setting `coords="minimal", data_vars="minimal"` in the call to `open_mfdataset` help?

**Reply by mdsumner** [reply]
ah thanks!  that gives me a lot to explore also

**Comment by mdsumner** [comment]
it does!

```python
import xarray as xr
## open two and crs is living on the time dimension
xr.open_mfdataset(['NSIDC0051_SEAICE_PS_S25km_20220522_v2.0.nc', 
                  'NSIDC0051_SEAICE_PS_S25km_20220523_v2.0.nc'], engine = 'netcdf4')
# ..
# Data variables:
#    crs         (time) |S1 b'' b''
#    F17_ICECON  (time, y, x) float32 dask.array<chunksize=(1, 332, 316), meta=np.ndarray>

## set minimal crs is properly hanging loose
xr.open_mfdataset(['NSIDC0051_SEAICE_PS_S25km_20220522_v2.0.nc', 
                  'NSIDC0051_SEAICE_PS_S25km_20220523_v2.0.nc'], engine = 'netcdf4', coords='minimal', data_vars='minimal')
# Data variables:
#    crs         |S1 b''
#    F17_ICECON  (time, y, x) float32 dask.array<chunksize=(1, 332, 316), meta=np.ndarray>               

```

awesome
